community/kernel/src/main/java/org/neo4j/kernel/impl/core/RelationshipProxy.java

```diff
@@ -458,13 +462,20 @@ public int compareTo( Object rel )
     @Override
     public boolean equals( Object o )
     {
-        return o instanceof Relationship && this.getId() == ((Relationship) o).getId();
+        if ( o instanceof Relationship )
+        {
+            return this.getId() == ((Relationship) o).getId();
+        }
+        else
+        {
+            return super.equals( o );
```

Comment:
This feels very dangerous - since it means `Relationship` objects might now be equal to other `RelationshipValue` objects.